src/ring_theory/hahn_series.lean

```diff
@@ -779,51 +787,55 @@ end
 section domain
 variables {Γ' : Type*} [ordered_cancel_add_comm_monoid Γ']
 
+lemma emb_domain_mul (f : Γ ↪o Γ') (hf : ∀ x y, f (x + y) = f x + f y) (x y : hahn_series Γ R) :
+  emb_domain f (x * y) = emb_domain f x * emb_domain f y :=
+begin
```

Comment:
The proof of this lemma is mostly unchanged, except now we can talk about `f` instead of `⟨⟨f, hfi⟩, hf⟩`